R-package/tests/testthat/test_basic.R

```diff
@@ -2780,3 +2780,78 @@ for (x3_to_categorical in c(TRUE, FALSE)) {
     })
   }
 }
+
+test_that("lightgbm() accepts objective as function argument and under params", {
+  bst1 <- lightgbm(
+    data = train$data
+    , label = train$label
+    , params = list(objective = "regression")
+    , nrounds = 5L
+    , verbose = -1L
+  )
+  expect_equal(bst1$params$objective, "regression")
+
+  bst2 <- lightgbm(
+    data = train$data
+    , label = train$label
+    , objective = "regression"
+    , nrounds = 5L
+    , verbose = -1L
+  )
+  expect_equal(bst2$params$objective, "regression")
+})
+
+test_that("lightgbm() prioritizes objective under params over objective as function argument", {
+  bst1 <- lightgbm(
+    data = train$data
+    , label = train$label
+    , objective = "regression"
+    , params = list(objective = "regression_l1")
+    , nrounds = 5L
+    , verbose = -1L
+  )
+  expect_equal(bst1$params$objective, "regression_l1")
+
+  bst2 <- lightgbm(
+    data = train$data
+    , label = train$label
+    , objective = "regression"
+    , params = list(loss = "regression_l1")
+    , nrounds = 5L
+    , verbose = -1L
+  )
+  expect_equal(bst2$params$objective, "regression_l1")
```

Comment:
As @StrikerRUS pointed out to me in https://github.com/microsoft/LightGBM/pull/5007#discussion_r807440458, just testing the value in the attribute `$params$objective` in the `Booster`  object in R doesn't fully test that the expected objective is actually passed through to training.

For all tests where you're testing the value of `$params$objective`, can you please also add checks like those I added in #5007?

https://github.com/microsoft/LightGBM/blob/2f27d4b226815016402f4633876790663a0ed8b5/R-package/tests/testthat/test_basic.R#L703-L710